### ISSUE TITLE
How to do magnetoresistance simulation?

### ISSUE THREAD

**Issue by himadri2024** [ubermag/help#143]
Dear all,

Could you please help me to do magnetoresistance simulation in ubermag.

Thank you.

Regards,
Himadri

**Comment by marijanbeg** [comment]
Hi @himadri2024 , thank you for your question and for using Ubermag. I transferred your issue to our help repository (https://github.com/ubermag/help). This is the repository we monitor regularly and where you can get support from others in the community. Therefore, we would encourage you to raise future issues in this repository.

What type of magnetoresistance do you want to compute?

**Comment by himadri2024** [comment]
Dear sir,

I would like to simulate anisotropic magnetoresistance.   In this, i am interested in how the resistance changes in a ferromagnet depending upon the field and distance from a magnetic object, which will be helpful for sensing application and sensor fabrication.

Thank you.
Regards,
Himadri

**Comment by marijanbeg** [comment]
Hi @himadri2024, thank you for the clarification. This is something that is not in Ubermag, but you can relatively simply compute it yourself as long as you assume a uniform current density vector.

Please try writing the code and we can help you on the way.

**Comment by marijanbeg** [comment]
Hi @himadri2024, we think it would be great if we could work on this together with you. We could start by you sharing the Jupyter notebook where you outline the requirements, share equations, and attempt computing the AMR. After that, we could exchange the notebook several times until we get the solution you need. Besides, we can identify if there are any functionalities we can move to Ubermag to make this computation easier, and we can use your notebook as a tutorial for other users and acknowledge you as a contributor to Ubermag.

**Comment by himadri2024** [comment]
Thank you, sir. I need some time. Then I will send you all details.

**Comment by himadri2024** [comment]
Dear sir,

I have tried to simulate the AMR curve using Zhang-Li STT by maintaining constant current density.

In order to get the AMR curve in the In-plane (IP) and out of plane (OOP) direction, the parallel component of magnetization (Mx) and the perpendicular component of magnetization (My) to the current direction are used. (Ref: J. Liang, et al. Applied Physics Letters 100.18 (2012): 182402)

I have attached the simulation file here.
[Magnetoresistance.zip](https://github.com/ubermag/help/files/6907331/Magnetoresistance.zip)

How can I send only charge current instead of spin current?

Please tell me whether the simulations are correct or not and help me to proceed further.

Thank you.
Regards,
Himadri

**Comment by MehranSedrpooshan** [comment]
Hi,

I was also wondering how to simulate magnetoresistance. I looked at the above code, thanks @himadri2024 for writing, and @marijanbeg for keeping this issue open.
Instead, I think we should consider blocks as a set of resistors parallel and in series. The resistance of each block is a function of the angle (θ) between the current and its magnetization (ρ(θ) = ρ﬩ + (ρ‖ - ρ﬩) cos^2 θ). By solving this "large circuit" for each applied magnetic field, one can achieve a magnetoresistance curve. 
However, I am not sure how to write this!

Regards

**Comment by himadri2024** [comment]
Yes, that's a great idea. We can consider a combination of resistors. 
If I am not wrong, we can define resistivity for the entire region or each cell. But how it will be included in the system energy so that the resistivity can be related.
We need help from @marijanbeg sir. It will be helpful in modeling GMR and TMR in the future. 

Regards,
himadri
